mesonbuild/backend/ninjabackend.py

```diff
@@ -1826,7 +1826,7 @@ def scan_fortran_module_outputs(self, target):
         if compiler is None:
             self.fortran_deps[target.get_basename()] = {}
             return
-        modre = re.compile(r"\s*module\s+(\w+)", re.IGNORECASE)
+        modre = re.compile(r"^(?<!sub)\s*module\s+(\w+)\s*$", re.IGNORECASE)
```

Comment:
How about using `\b` to indicate a word boundary at the beginning?

Reply:
yes thanks that's better